docs/maintenance/upgrade.md

```diff
@@ -2,7 +2,12 @@
 
 To upgrade Codacy to the latest stable version:
 
-1.  Store all your currently defined configuration values in a file:
+1.  Check the [release notes](/release-notes/) for all Codacy Self-hosted versions between your current version and the most recent version for breaking changes and follow the instructions provided carefully.
```

Comment:
If we guarantee that we only add breaking changes when releasing major versions, we can make that explicit here.